### ISSUE TITLE
Query is read-only in DBeaver 23.0.0, but editable in 22.2.2

### ISSUE THREAD

**Issue by xoelop** [dbeaver/dbeaver#19376]
### Description

The result set from a query that includes a primary key (the `id` column) is not editable in DBeaver 23.0.0

![image](https://user-images.githubusercontent.com/8508402/225292692-21520fd2-50a0-4ea6-9a5c-a63a09616a89.png)

Note the lock in the top left corner of the result set showing that the results are not editable.

However, the same query is editable in DBeaver 22.2.2. 

![image](https://user-images.githubusercontent.com/8508402/225293544-69b45d76-ca7c-4c83-908b-729d05e837b7.png)

Using Postgres 15

### DBeaver Version

Community Edition 23.0.0

### Operating System

macOS Ventura 13.1

### Database and driver

Postgres 15

### Steps to reproduce

Text of the query:

```sql
SELECT
	k.id
	, k.name
	, IGNORE
	, case_insensitive
	, regex_pattern 
	, lower(k.name) IN (SELECT lower(name) FROM word) is_english_word
	, count(*) OVER ()
	, ts.jobs  
FROM keyword k
JOIN technologies_summary ts 
ON k.id = ts.id
WHERE IGNORE IS FALSE
AND lower(k.name) IN (SELECT lower(name) FROM word) IS TRUE 
AND case_insensitive IS TRUE 
```

**Comment by LonwoLonwo** [comment]
Reproducible

![2023-03-15 15_58_10-DBeaver Ultimate 23 0 0 - _postgres_ Script-110](https://user-images.githubusercontent.com/45152336/225315736-fec1b97b-1e12-4da6-aee3-68454c8fee9b.png)

Thanks for the report @xoelop

**Comment by LonwoLonwo** [comment]
These cases work for me:

```sql
SELECT
	k.id
	, k.name
	, k."ignore"
	, k.case_insensitive
	, k.regex_pattern 
	, lower(k.name) IN (SELECT lower(name) FROM word) is_english_word
	, count(*) OVER ()
	, ts.jobs  
FROM keyword k
JOIN technologies_summary ts 
ON k.id = ts.id
```

```sql
SELECT
	k.*,
	ts.* 
FROM keyword k
JOIN technologies_summary ts 
ON k.id = ts.id
WHERE "ignore" IS FALSE
--AND lower(k.name) IN (SELECT lower(name) FROM word) IS TRUE 
AND case_insensitive IS TRUE
```

```sql
SELECT
	k.id
	, k.name
	, k."ignore"
	, case_insensitive
	, regex_pattern 
	, lower(k.name) IN (SELECT lower(name) FROM word) is_english_word
	, count(*) OVER ()
	, ts.jobs  
FROM keyword k
JOIN technologies_summary ts 
ON k.id = ts.id
WHERE "ignore" IS FALSE
--AND lower(k.name) IN (SELECT lower(name) FROM word) IS TRUE 
AND case_insensitive IS TRUE 
```

So, the issue is probably in the "IGNORE" keyword (need quotes) and the condition `AND lower(k.name) IN (SELECT lower(name) FROM word) IS TRUE `

**Comment by xoelop** [comment]
Indeed, those 2 things you pointed out did it. The "ignore" I can fix by wrapping it with quotes, but no idea how to rewrite the other condition. Any ideas?

**Comment by andey-k** [comment]
based on the doc you still have a few words that's in the keyword dictionary.
the guide recommends to add quotes or include the AS keyword in known cases or as an insurance against future ambiguity.

lower(name) as name
count ... as count
k.name as name
from word as word

which makes me think if there is a way to declare the standard (sql:92 etc) the code is supposed to conform to (something like option explicit in vba) or set it in as a menu option so that the rules can be properly applied and even hint where it doesn't confirm to (user can disable the hint).

https://www.postgresql.org/docs/current/sql-keywords-appendix.html
https://www.postgresql.org/docs/current/queries-select-lists.html#QUERIES-COLUMN-LABELS
